src/evaluators/CallExpression.js

```diff
@@ -79,6 +100,21 @@ function evaluateReference(
     if (base.kind === "conditional") {
       return evaluateConditionalReferenceBase(ref, ast, strictCode, env, realm);
     }
+    let referencedName = ref.referencedName;
+
+    // When dealing with a PrimitiveValue, like StringValue, NumberValue, IntegralValue etc
+    // if we are referencing a prototype method, then it's safe to access, even
+    // on an abstract value as the value is immutable and can't have a property
+    // that matches the prototype method (unless the prototype was modified).
+    // In pure scope, we assume the global prototype of built-ins has not been altered.
```

Comment:
I was not aware of this, can we document this somewhere in the code or the wiki for pure evaluation mode?

Reply:
It should be documented on https://github.com/facebook/prepack/wiki/Compiler-assumptions